crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -729,12 +767,36 @@ impl Room {
 
         let heroes = members
             .into_iter()
-            .filter(|u| u.user_id() != self.own_user_id)
+            .filter(heroes_filter)
             .take(NUM_HEROES)
             .map(|u| u.name().to_owned())
             .collect();
 
-        Ok((heroes, num_joined_invited))
+        trace!(
+            ?heroes,
+            num_joined_invited,
+            num_service_members,
+            "Computed a room summary since we didn't receive one."
+        );
+
+        Ok((heroes, num_joined_invited as u64))
+    }
+
+    async fn get_member_hints(&self) -> StoreResult<MemberHintsEventContent> {
+        Ok(self
+            .store
+            .get_state_event_static::<MemberHintsEventContent>(self.room_id())
+            .await?
+            .and_then(|event| {
+                event
+                    .deserialize()
+                    .inspect_err(|e| warn!("Couldn't deserialize the member hints event: {e}"))
+                    .ok()
+            })
+            .and_then(|event| as_variant!(event, SyncOrStrippedState::Sync))
+            .and_then(|event| as_variant!(event, SyncStateEvent::Original))
+            .map(|e| e.content)
```

Comment:
(light suggestion) A bit more concise, and avoids the chaining of `and_then` twice + `map`:

```rust
            .and_then(|event| as_variant!(event, SyncOrStrippedState::Sync(SyncStateEvent::Original(e)) => e.content))
```

Reply:
3c12242e22fce2996410f2f4d75dd732c9fa5bd8